lib/SessionManager.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Created by elad.benedict on 9/6/2015.
+ */
+
+var persistenceFormat = require('./PersistenceFormat');
+var glob = require('glob');
+var config = require('./Configuration');
+var Q = require('q');
+var qio = require('q-io/fs');
+var _ = require('underscore');
+
+module.exports = (function(){
+
+    var isNewSession = function isNewSession(entryId) {
+        return Q.fcall(function () {
+            var entryFolder = persistenceFormat.getEntryFullPath(entryId);
+            return entryFolder;
+        }).then(function (entryFolder) {
+            var getManifestFiles = Q.denodeify(glob);
+            return getManifestFiles("**/*.m3u8", {cwd: entryFolder, realpath: true});
```

Comment:
You're right, however in this (session manager's) context I don't know what flavors are available for the entry.
I could have passed those as part of the session manager's API, however I think the current API (accepting only an entry id) is more sound (For example, when we'll change the logic to infer a new session based on backend entry data it won't have to change)